Model/AdyenInitPayments.php

```diff
@@ -189,6 +223,11 @@ protected function buildPaymentsRequest(Quote $quote, array $stateData): array
         );
         $currency = $quote->getQuoteCurrencyCode();
         $amount = $quote->getSubtotalWithDiscount();
+        $paymentMethodCode = "adyen_{$stateData['paymentMethod']['type']}";
+        $userType = $this->userContext->getUserType();
+        $customerId = $this->userContext->getUserId();
+        $isLoggedIn = (int)$userType === UserContextInterface::USER_TYPE_CUSTOMER;
```

Comment:
Integer casting is irrelevant here. [getUserType()](https://github.com/magento/magento2/blob/4de008a9e3ae4f7e627ce36eb407d9ab5c6aafe6/app/code/Magento/Authorization/Model/UserContextInterface.php#L38C21-L38C32) method returns `?int`. Even it's `null`, type-strict comparison `===` will give the correct result.